Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSSingleReduction2.py

```diff
@@ -0,0 +1,530 @@
+        final_output_workspace_names = {}
+        for reduction_mode, output_bundles in reduction_mode_vs_output_bundles.items():
+            # Find the sample in the data collection
+            state, reduction_mode = next(((output_bundle.state, output_bundle.reduction_mode)
+                                          for output_bundle in output_bundles
+                                          if output_bundle.data_type == DataType.Sample), None)
+
+            # Get the workspace name
+            name = self._get_output_workspace_name(state, reduction_mode=reduction_mode)
+            final_output_workspace_names.update({reduction_mode: name})
+
+        return final_output_workspace_names
+
+    def _get_merged_workspace_name(self, output_parts_bundle):
+        """This method gets the output workspace names for a merged bundle. This only occurs
+         if the reduction mode is Merged.
+         :param output_parts_bundle: a list of OutputBundles containing workspaces for a single event slice.
+         :return: a workspace name
+         """
+        state = output_parts_bundle[0].state
+        return self._get_output_workspace_name(state, reduction_mode=ReductionMode.Merged)
+
+    def _get_output_workspace_name(self, state, reduction_mode=None, data_type=None,
+                                   can=False, sample=False, transmission=False, fitted=False):
+        """
+        Get the output names for the sliced workspaces (within the group workspaces, which are already named).
+
+        :param state: a SANS state object
+        :param reduction_mode: an optional ISISReductionMode enum: "HAB", "LAB", "Merged", or "All"
+        :param data_type: an optional DataType enum: "Sample" or "Can"
+        :param can: optional bool. If true then creating name for a can workspace
+        :param sample: optional bool. If true then creating name for a sample workspace. Sample and can cannot both be
+                       true
+        :param transmission: optional bool. If true then creating name for a transmission workspace
+        :param fitted: optional bool. If true then workspace is a fitted transmission workspace, otherwise unfitted
+        :return: name of the workspace
+        """
+        _multi = {"event_slice": True,
+                  "period": self.getProperty("Period").value,
+                  "wavelength_range": self.getProperty("WavelengthRange").value}
+
+        if not transmission:
+            _suffix = ""
+            if can:
+                if reduction_mode == ISISReductionMode.HAB:
+                    _suffix = "_hab_can"
+                elif reduction_mode == ISISReductionMode.LAB:
+                    _suffix = "_lab_can"
+            elif sample:
+                if reduction_mode == ISISReductionMode.HAB:
+                    _suffix = "_hab_sample"
+                elif reduction_mode == ISISReductionMode.LAB:
+                    _suffix = "_lab_sample"
```

Comment:
this section can be simplified slightly to require less if
```
suffix_end = "_{}_can" if can else "_{}_sample"
if reduction_mode == ISISReductionMode.HAB:
	_suffix = suffix_end.format("hab")
elif reduction_mode == ISISReductionMode.LAB:
	_suffix = suffix_end.format("lab")
```

Reply:
Good idea. will change